src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -0,0 +1,197 @@
+            }
+
+            requestStream.Flush();
+            requestStream.Close();
+
+            UploadProgress?.ChangeState(UploadState.PendingResponse);
+            var response = await request.GetResponseAsync();
+            response.Dispose();
+
+            UploadProgress?.ChangeState(UploadState.Done);
+        }
+
+        private async Task <string> CreateNestedFoldersIfNeeded(string folderName)
+        {
+            ExtractUrlAndDirectories(out string url, out List<string> directories);
+
+            // create the nested folders including the new folder
+            directories.Add(folderName);
+
+            foreach (var directory in directories)
+            {
+                url += $"/{directory}";
+                var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.MakeDirectory, keepAlive: false);
+
+                try
+                {
+                    var response = await request.GetResponseAsync();
+                    response.Close();
+                }
+                catch (WebException e)
+                {
+                    var response = (FtpWebResponse)e.Response;
+                    if (response.StatusCode == FtpStatusCode.ActionNotTakenFileUnavailable)
+                    {
+                        // folder already exists
+                        continue;
+                    }
+
+                    throw;
+                }
+            }
+
+            return url;
+        }
+
+        private void ExtractUrlAndDirectories(out string url, out List<string> dirs)
+        {
+            var address = Regex.Match(_url, @"^(ftp://)?(\w*|.?)*/").Value.Replace("ftp://", "").Replace("/", "");
+            dirs = Regex.Split(_url.Replace(address, "").Replace("ftp://", ""), "/").Where(x => x.Length > 0).ToList();
+            address = $"ftp://{address}";
+            var uri = new Uri(address);
+            var port = uri.Port > 0 ? uri.Port : (_port ?? DefaultFtpPort);
+
+            if (port < 1 || port > 65535)
+                throw new ArgumentException("Port number range: 1-65535");
+
+            url = $"ftp://{uri.Host}:{port}";
+        }
+
+        private FtpWebRequest CreateFtpWebRequest(string url, string method, bool keepAlive)
+        {
+            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
+            request.Method = method;
+
+            request.Credentials = new NetworkCredential(_userName, _password);
+            request.EnableSsl = _useSsl;
+            if (_useSsl)
+            {
+                var byteArray = Convert.FromBase64String(_certificateAsBase64);
+
+                try
+                {
+                    var x509Certificate = new X509Certificate(byteArray);
+                    request.ClientCertificates = new X509CertificateCollection(new[] { x509Certificate });
+                }
+                catch (Exception e)
+                {
+                    throw new ArgumentException("This is not a valid certicate file!", e);
```

Comment:
Here you can use the certificate file name, but it may be confusing for the user if they are looking at file name, though.